### ISSUE TITLE
Analyzer assumes anything with the same name as the class is a constructor.

### ISSUE THREAD

**Issue by lrhn** [dart-lang/sdk#48334]
Example from a test I made:
```dart
enum ConflictClassStatic {
  e1;
  static int ConflictClassStatic() => 37;
//^^^^^^
// [analyzer] SYNTACTIC_ERROR.STATIC_CONSTRUCTOR
//       ^^^
// [analyzer] SYNTACTIC_ERROR.CONSTRUCTOR_WITH_RETURN_TYPE
//           ^^^^^^^^^^^^^^^^^^^
// [analyzer] COMPILE_TIME_ERROR.NON_CONST_GENERATIVE_ENUM_CONSTRUCTOR
//                                 ^^^^^^
// [analyzer] COMPILE_TIME_ERROR.RETURN_IN_GENERATIVE_CONSTRUCTOR
//                                    ^^
// [analyzer] COMPILE_TIME_ERROR.RETURN_OF_INVALID_TYPE
}
```
The error here is not that the constructor fails to be a valid constructor in all ways. It's not a constructor.
You are not allowed to have a static member with the same name as the class, that's the only error.

So, maybe, there should be a heuristic to recognize something as a badly named member instead of a completely malformed constructor. That would likely improve the error messages.

**Comment by bwilkerson** [comment]
I haven't looked at the code to confirm, but I suspect that the heuristic to recover by considering this to be a constructor that's incorrectly prefixed by the keyword `static` is in the parser. (That's where most syntactic errors are generated.)

**Comment by scheglov** [comment]
Yes, looking into the parser, any method with the name of the enclosing declaration is considered to be a constructor.
